### PR TITLE
[SPARK-37727][SQL][TESTS][FOLLOW-UP] Add test cases for logs in SparkSession.builder.getOrCreate

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR adds some test cases for https://github.com/apache/spark/pull/35001. Test cases were not added because there are related bugs but they are fixed now.

### Why are the changes needed?

To avoid regressions.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

This PR adds test cases that were manually tested in https://github.com/apache/spark/pull/35001.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @AngersZhuuuu @yaooqinn FYI

**Review by AngersZhuuuu — COMMENTED** [review]
LGTM.

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 1]
```diff
@@ -486,4 +487,83 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
+
+  test("SPARK-37727: Show ignored configurations in debug level logs") {
+    // Create one existing SparkSession to check following logs.
+    SparkSession.builder().master("local").getOrCreate()
+
+    val logAppender = new LogAppender
+    logAppender.setThreshold(Level.DEBUG)
+    withLogAppender(logAppender, level = Some(Level.DEBUG)) {
+      SparkSession.builder()
+        .master("local")
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:499]:
```suggestion
```

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 2]
```diff
@@ -486,4 +487,83 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
+
+  test("SPARK-37727: Hide runtime SQL configurations in logs") {
+    // Create one existing SparkSession to check following logs.
+    SparkSession.builder().master("local").getOrCreate()
+
+    val logAppender = new LogAppender
+    logAppender.setThreshold(Level.DEBUG)
+    withLogAppender(logAppender, level = Some(Level.DEBUG)) {
+      // Ignore logs for runtime SQL configurations
+      SparkSession.builder().master("local").config("spark.sql.ansi.enabled", "true").getOrCreate()
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:550]:
```suggestion
      SparkSession.builder().config("spark.sql.ansi.enabled", "true").getOrCreate()
```

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 3]
```diff
@@ -486,4 +487,83 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
+      // Show logs for Spark core configuration
+      SparkSession.builder().master("local").config("spark.buffer.size", "1234").getOrCreate()
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:552]:
```suggestion
      SparkSession.builder().config("spark.buffer.size", "1234").getOrCreate()
```

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 4]
```diff
@@ -486,4 +487,83 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
+      // Show logs for custom runtime options
+      SparkSession.builder().master("local").config("spark.sql.source.abc", "abc").getOrCreate()
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:554]:
```suggestion
      SparkSession.builder().config("spark.sql.source.abc", "abc").getOrCreate()
```

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 5]
```diff
@@ -486,4 +487,83 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach wit
...
+      // Show logs for static SQL configurations
+      SparkSession.builder().master("local").config("spark.sql.warehouse.dir", "xyz").getOrCreate()
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:556]:
```suggestion
      SparkSession.builder().config("spark.sql.warehouse.dir", "xyz").getOrCreate()
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
+1，LGTM

**Review by dongjoon-hyun — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Wait .. the tests pass in IDE but fails in SBT shall command. Let me investigate this a bit more.

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala [hunk 6]
```diff
@@ -249,6 +249,7 @@ abstract class SparkFunSuite
       if (level.isDefined) {
         logger.setLevel(level.get)
         logger.get().setLevel(level.get)
+        LogManager.getContext(false).asInstanceOf[LoggerContext].updateLoggers()
```
Comment (HyukjinKwon) [inline core/src/test/scala/org/apache/spark/SparkFunSuite.scala:252]: @viirya, I borrowed your fix from https://github.com/apache/spark/pull/35012 but TBH I have less background here. Do you mind reviewing this please? This change fixes the issue in `withLogAppender` where the logger level is not changed.

Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:252]: I think the issue is the same. Once the log level is set, it doesn't get changed. That's why it passed in IDE when I run it individually, and it fails when I run them in batch via SBT.

Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:252]: I don't mind making this in a separate PR if you guys feel that's better.

Comment (viirya) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:252]: lgtm. I thought we should update the loggers here too. I actually was wondering if this should be do the same, but as no test failure happens I thought it maybe fine. :)

**Review by viirya — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala (hunk 6 above)
Comment (HyukjinKwon) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:252]: Thanks!

**Comment by HyukjinKwon** [comment]
Merged to master!

**Pull request closed without merging**